src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,589 @@
+                reportInterval=[3, 3],
+                autoResubscribe=False
+            )
+        except ChipStackError as e:
+            # Verify that the DUT returns an "INVALID_ACTION" status response
+            asserts.assert_equal(e.err, INVALID_ACTION_ERROR_CODE,
+                                 "Incorrect error response for subscription to unallowed node")
+
+        '''
+        ##########
+        Step 7
+        ##########
+        '''
+        self.print_step(7, "CR1 sends a subscription request action for an attribute with an empty DataVersionFilters field. DUT sends a report data action with the data of the attribute along with the data version. Tear down the subscription for that attribute. Start another subscription with the DataVersionFilter field set to the data version received above.")
+
+        # Subscribe to attribute with empty dataVersionFilters
+        sub_cr1_empty_dvf = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=node_label_attr_path,
+            keepSubscriptions=False
+        )
+
+        # Verify DataVersion attribute data came back
+        self.verify_attribute_data(
+            sub=sub_cr1_empty_dvf,
+            cluster=Clusters.BasicInformation,
+            attribute=Clusters.Attribute.DataVersion
+        )
+
+        # Get DataVersion
+        data_version = self.get_attribute_from_sub_dict(
+            sub=sub_cr1_empty_dvf,
+            cluster=Clusters.BasicInformation,
+            attribute=Clusters.Attribute.DataVersion
+        )
+        data_version_filter = [(0, Clusters.BasicInformation, data_version)]
+
+        # Subscribe to attribute with provided DataVersion
+        sub_cr1_provided_dvf = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=node_label_attr_path,
+            reportInterval=(10, 20),
+            keepSubscriptions=False,
+            dataVersionFilters=data_version_filter
+        )
+
+        # Verify that the subscription is activated between CR1 and DUT
+        asserts.assert_true(sub_cr1_provided_dvf.subscriptionId, "Subscription not activated")
+
+        sub_cr1_provided_dvf.Shutdown()
+
+        '''
+        ##########
+        Step 8
+        ##########
+        '''
+        self.print_step(8, "CR1 sends a subscription request action for an attribute and sets the MinIntervalFloor value to be same as MaxIntervalCeiling. Activate the Subscription between CR1 and DUT. Modify the attribute which has been subscribed to on the DUT.")
+
+        # Subscribe to attribute
+        sub_cr1_update_value = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=node_label_attr_path,
+            reportInterval=(same_min_max_interval_sec, same_min_max_interval_sec),
+            keepSubscriptions=False
+        )
+
+        # Modify attribute value
+        new_node_label_write = "NewNodeLabel_" + str(random.randint(1000, 9999))
```

Comment:
please explain in a comment why random is used.
If this code is run on a device with a known label, we can pick anything.
If this code is run on devices with unknown labels, do we care about collisions?

Reply:
Removed random